### ISSUE TITLE
Sound Visualisation modes won't work

### ISSUE THREAD

**Issue by GithubusMaximus** [psieg/Lightpack#194]
Hello, I'm having issues with Sound Visualisation modes with Prismatik 5.11.2.17 rev 1863b15f that runs with Adalight device at 500000baudrate.

Simply when i pick the mode to be "Sound Visualisation" nothing really happens. Tried both loopback and no loopback mode and waited several minutes for reaction. In the "animate colours" mode the LEDs are stuck on first colour all the time. In "Change colour with rate" they stay turned off all the time.

Standard screen grabbing mode works flawlessly. I have other programs on PC that use "audio visualisation" capabilities and they work just fine.

Don't really know how to help troubleshooting. Windows 10 Pro 64 bit 1709. Motherboard is Asus Maximus X hero with realtek ALC1220 (S1220) and using headphones (that are my only playback device).

**Comment by psieg** [comment]
The loopback device should carry the signal your speakers output. Can you get it to flash using a microphone and the corresponding non-loopback device?

There are different ways to get the sound output, and which one works best depends primariliy on the audio driver. If you see a loopback device however, that should in theory work. Do you see that loopback device in the audio control panel? It might be muted or something.

**Comment by GithubusMaximus** [comment]
Hello,

I can get it to flash when using microphone. It does not flash when using Speakers (Realtek High Definition Audio). Neither when using Speakers (Realtek High Definition Audio) Loopback.

I attached screenshot of my audio devices, which one is supposed to be loopback?

![1](https://user-images.githubusercontent.com/37982341/45892640-f9642980-bdc8-11e8-8e7c-3643178164a0.PNG)
![2](https://user-images.githubusercontent.com/37982341/45892641-f9642980-bdc8-11e8-82b6-3a71a19b8b80.PNG)

**Comment by GithubusMaximus** [comment]
Is the loopback device something that should get installed during Prismatik installation?

**Comment by GithubusMaximus** [comment]
I just found Prismatik logs, noticed something in there:

> 19:58:15:115 Critical: void __cdecl SoundManager::start(bool) Can't initialize WASAPI device 6
[Prismatik.0.log](https://github.com/psieg/Lightpack/files/2408976/Prismatik.0.log)

**Comment by GithubusMaximus** [comment]
Here is a log when audio visualisation was set as an active mode upon close and it remains active upon restart of Prismatik
[Prismatik.0.log](https://github.com/psieg/Lightpack/files/2408979/Prismatik.0.log)

**Comment by psieg** [comment]
No the loopback device either exists or doesn't depending on your driver. It may not be visible in the sound settings (mine look like yours). 
The error code 6 is BASS_ERROR_FORMAT, i.e. your sound driver doesn't support the format. 

Try this version, it uses an additional flag and report back please.
[Prismatik_auto_format_test.zip](https://github.com/psieg/Lightpack/files/2425808/Prismatik_auto_format_test.zip)

**Comment by GithubusMaximus** [comment]
I just swapped the Prismatik.exe 

Here are two logs. In one i opened it as usual, switched to audio visualisation mode, then to loopback device and to audio mode again.

Second log is when i shut down Prismatik already in audio mode and started it like that.

Audio visualisation wouldn't work.

[Prismatik.0.log](https://github.com/psieg/Lightpack/files/2428737/Prismatik.0.log)
[Prismatik.1.log](https://github.com/psieg/Lightpack/files/2428738/Prismatik.1.log)

**Comment by psieg** [comment]
hmm, the errors message remains the same `Can't initialize WASAPI device 6`.
I'm sorry I don't know how to fix this, are you using a fancy sound card or something? Realtek HD Audio seems quite normal...

**Comment by GithubusMaximus** [comment]
> 
> 
> hmm, the errors message remains the same `Can't initialize WASAPI device 6`.
> I'm sorry I don't know how to fix this, are you using a fancy sound card or something? Realtek HD Audio seems quite normal...

I don't know if its fancy, its integrated audio based on Realtek S1220, pretty common. Using the legacy drivers (HD audio drivers). In some days from now ill be switching to UMP based drivers and ill see if it changes anything.

**Comment by pabloariel01** [comment]
try disabling the sound devices in the device manager and then start reenabling one by one, i couldnt make it work until i found out there was an nvidea controller blocking it

**Comment by szuvi** [comment]
I disabled all the sound output devices except the main I am using. Then in Pirsmatic I choose my output device with loopback in parenthesies. This solution worked for me.

**Comment by Piyush36** [comment]
I am facing the exact same problem. Whenever I use two output devices. Headphones and speakers. It is working perfectly for my headphones but when I switch to the speakers, it doesn't work. Getting the same error in logs.
22:28:56:235 Critical: void __cdecl WindowsSoundManager::start(bool) Can't initialize WASAPI device 6
I also disabled all other playback devices from device manager as well but still it is not working.

**Comment by zomfg** [comment]
can you try [this](https://github.com/zomfg/Lightpack/releases/download/5.11.2.24/Parismatik_win_wasapi.zip)

**Comment by Piyush36** [comment]
same 
00:33:12:982 Critical: void __cdecl WindowsSoundManager::start(bool) Can't initialize WASAPI device 6

**Comment by Piyush36** [comment]
[Prismatik.0.log](https://github.com/psieg/Lightpack/files/5668103/Prismatik.0.log)
attaching full file for incase it is helpful

**Comment by zomfg** [comment]
redownload, rerun, show the wasapi line again

**Comment by Piyush36** [comment]
Wow! You are changing the base code debug lines. Thanks man.

00:49:47:377 Debug: void __cdecl WindowsSoundManager::start(bool) true
00:49:47:407 Critical: void __cdecl WindowsSoundManager::start(bool) Can't initialize WASAPI device 8 chans 0 format 0 freq 0
00:49:47:407 Debug: void __cdecl SoundManagerBase::initColors(int) 100

**Comment by zomfg** [comment]
hmmm did you get other errors this time?
can you re-run til you get `...device 6...` thing (`8` says there was another problem before that point, but thats different)

**Comment by Piyush36** [comment]
[Prismatik.0.log](https://github.com/psieg/Lightpack/files/5668268/Prismatik.0.log)
This is is weird. Now whenever I am switching to speaker (for which the issue is) it is printing device 8.
01:14:31:015 Critical: void __cdecl WindowsSoundManager::start(bool) Can't initialize WASAPI device 8 chans 0 format 0 freq 0

**Comment by Piyush36** [comment]
I will be more than happy to give you access to the system even if you want.

**Comment by zomfg** [comment]
nah I don't have more tricks unfortunately, but..
the `6` (you can disregard `8`) means unsupported sound format (not to be confused with sound file format etc)
so I would try everything in this window
![image](https://user-images.githubusercontent.com/239811/101680378-eea51600-3a60-11eb-8b5d-89dda80ba191.png)

**Comment by Piyush36** [comment]
I have set format as same as headphones since headphones are working fine (16 bit 48000Hz). But yeah, as you said, they are a solution of 6. But I am getting 8. Any clue about 8?

**Comment by zomfg** [comment]
the `8` is probably just a side effect of me trying to get more info at the place, so switch back to normal release and you'll get only 6s I guess
on my system everything from 16bit / 44.1k to 32bit / 192k works fine

**Comment by Piyush36** [comment]
hahah. yeah just thought of. I installed previous version. Now I am getting 6 only .
Quick question. I am checking each one of them from the dropdown. Should I select one by one then after each one I change something (turn on/off or switch sound device inside the Prismatik) or just changing the format one by one checking lights would do?

**Comment by zomfg** [comment]
I would relaunch Prismatik just in case
take your time

**Comment by Piyush36** [comment]
Nope, not working. Guess I will quit. But comment here in future if you find anything about it.
Thanks for your help so far.

**Comment by Piyush36** [comment]
Update: I found a workaround. I looked up on how can I output sound on two devices. So basically before headphones were working audio sync. I have included speaker in output audio as well. Its working now. XD

**Comment by laresistenciadelbit** [comment]
> 
> 
> nah I don't have more tricks unfortunately, but..
> the `6` (you can disregard `8`) means unsupported sound format (not to be confused with sound file format etc)
> so I would try everything in this window
> ![image](https://user-images.githubusercontent.com/239811/101680378-eea51600-3a60-11eb-8b5d-89dda80ba191.png)

This really helped! Thank you.

**Comment by yusuff34** [comment]
![Ekran Alıntısı](https://user-images.githubusercontent.com/46285794/107859701-48083780-6e4c-11eb-8ffa-1eff59e3a854.PNG)

the solution of the problem is click away, it is fix

**Comment by kakaopiraten** [comment]
I had same problem. 2 devices. 1 headphone, 2 monitors.
What worked for me was to configure the device and pick 7.1 Surround instead of Stereo.

![image](https://user-images.githubusercontent.com/55191176/113177606-0261cb80-924e-11eb-969a-4ff3cfa5e027.png)
